lib/confluence/vendor_specific.es6.js

```diff
@@ -0,0 +1,136 @@
+  name: 'VendorSpecific',
+  extends: 'org.chromium.apis.web.MetricComputer',
+  documentation: `VendorSpecific is a class that computes the Vendor Specific
+    API metrics from browserApiDAO. It shows the number of APIs
+    that only available in this browser.`,
+  requires: [
+    'org.chromium.apis.web.Browser',
+    'org.chromium.apis.web.WebInterface',
+    'org.chromium.apis.web.BrowserWebInterfaceJunction',
+    'org.chromium.apis.web.VendorSpecificData',
+    'foam.mlang.ExpressionsSingleton',
+    'foam.dao.ArraySink',
+    'foam.dao.EasyDAO',
+  ],
+  properties: [
+    {
+      name: 'vendorSpecificDAO',
+      documentation: `This is a DAO that contains vendor specific data.`,
+      final: true,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'vendorSpecificDAO',
+          of: this.VendorSpecificData,
+          daoType: 'MDAO',
+        });
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'compute',
+      documentation: `Compute vendor specific API value for each browser in
+           browsers at the given date.`,
+      args: [
+        {
+          name: 'browsers',
+          typeName: 'Array of Browser',
+          documentation: `An array of brwsers. These browser are either
+              released at given date, or it is the most recent release
+              before given date.`,
+        },
+        {
+          name: 'date',
+          typeName: 'Date',
+          documentation: `The vendor specific value will be calculated at
+              given date.`,
+        },
+      ],
+      code: function(browsers, date) {
+        for (let i = 0; i < browsers.length; i++) {
+          let browser = browsers[i];
+          let year = date.getFullYear();
+          let month = date.getMonth();
+          let day = date.getDate();
+          let previousYear = new Date(year - 1, month, day);
+          let browserKeys = browsers.map((browser) => browser.browserKey);
+          // Find all browsers that
+          //   - a previous version of the browser.
+          //   - some other browser but released in previous one year range.
+          //   - if no release in one year, uese most recent browser.
+          this.browserDAO.where(this.mlang.OR(
+            this.mlang.AND(
+              this.mlang.IN(this.Browser.BROWSER_KEY, browserKeys)),
+            this.mlang.AND(
+              this.mlang.EQ(this.Browser.BROWSER_NAME, browser.browserName),
+              this.mlang.EQ(this.Browser.OS_NAME, browser.osName),
+              this.mlang.GT(this.Browser.RELEASE_DATE, previousYear),
+              this.mlang.LT(this.Browser.RELEASE_DATE, date)),
+            this.mlang.AND(
+              this.mlang.GT(this.Browser.RELEASE_DATE, previousYear),
+              this.mlang.LT(this.Browser.RELEASE_DATE, date))))
+          .select().then((result) => {
+            let browserIface = {};
+            let numPrevReleases = 1;
+            let allIfaces = {};
```

Comment:
Perhaps rename to `allOtherIfaces` to connote "all interfaces from other vendors"?